### PR TITLE
Tabs Component: Add tests for the tabs component

### LINKED ISSUE #319: Add tests for the tab component
Add tests for the Tab component

### PR DESCRIPTION
**Pull request by ritikamotwani**
Fixes: #319

### CODE REVIEW

**Comment by tingtingdisco** [comment]
nice, only the prop `onTabClick` is not tested

Screen Shot 2020-05-05 at 11 29 34 AM

you can run `yarn coverage:src`

and go to the coverage folder

Screen Shot 2020-05-05 at 11 29 59 AM

open the index,html in browser, to see which lines are not tested.

File: src/Display/Tabs/Tabs.test.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      )
+      .toJSON();
+    expect(snapshot).toMatchSnapshot();
+  });
+});
+
+describe('<Tabs/> should render correctly with the alignment prop', () => {
+  test('horizontal, when alignment is set horizontal', () => {
+    const { getByRole } = render(
+      <Tabs alignment="horizontal">
```
Comment (tingtingdisco) [inline src/Display/Tabs/Tabs.test.tsx]: I suggest testing all the possible `alignment`s, you can check `Divider.test.tsx`, import `jest-styled-components`, and generate snapshots with css included, so we don't need to check individual style. Just remember to check the snapshot and validate the css is being correctly generated.

File: src/Display/Tabs/Tabs.test.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    );
+    const tabContainer = getByRole('tablist');
+    expect(tabContainer).toHaveStyle('display: flex');
+  });
+});
+
+describe('<Tabs/> should render correctly with the variant prop', () => {
+  test('colored, when alignment is set colored', () => {
+    const { getByRole } = render(
+      <Tabs alignment="horizontal" variant={EHorizontalTabVariant.COLORED}>
```
Comment (tingtingdisco) [inline src/Display/Tabs/Tabs.test.tsx]: same here, test all variants

File: src/Display/Tabs/Tabs.test.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    fireEvent.click(tab[clickedTabIndex]);
+    const Job = queryByText('Software Engineering');
+    const Company = queryByText('Glints');
+
+    expect(tab[firstTabIndex]).not.toHaveStyle('font-weight: bold');
+    expect(tab[clickedTabIndex]).toHaveStyle('font-weight: bold');
+    expect(Job).not.toBeInTheDocument();
+    expect(Company).toBeVisible();
+  });
+});
```
Comment (tingtingdisco) [inline src/Display/Tabs/Tabs.test.tsx:111]: need one test case to validate `onTabClick` is called when click tab.

File: src/Display/Tabs/Tabs.test.tsx (hunk 1 above)
Comment (ritikamotwani) [reply src/Display/Tabs/Tabs.test.tsx]: This is so much better than testing appropriate styes individually. Thanks! Updated now.

File: src/Display/Tabs/Tabs.test.tsx (hunk 2 above)
Comment (ritikamotwani) [reply src/Display/Tabs/Tabs.test.tsx]: For the tabs, there's no UI for vertical tabs with colored variants, so haven't tested that combination. 
The colored variant exists only for the horizontal one.

File: src/Display/Tabs/Tabs.test.tsx (hunk 3 above)
Comment (ritikamotwani) [reply src/Display/Tabs/Tabs.test.tsx:111]: Added the check if onTabClick is called.

**Comment by ritikamotwani** [comment]
@tingtingdisco , updated this one.

**Review by tingtingdisco — APPROVED** [review]
LGTM thank u @ritikamotwani

**Pull request merged**